File: app/src/main/java/com/github/willjgriff/ethereumwallet/di/AppInjector.java
package com.github.willjgriff.ethereumwallet.di;

import android.content.Context;

import com.github.wiljgriff.ethereumwallet.di.*;
import com.github.wiljgriff.ethereumwallet.di.AppComponent;
import com.github.wiljgriff.ethereumwallet.di.modules.AppModule;

/**
 * Created by Will on 29/01/2017.
 */

public enum AppInjector {

	INSTANCE;

	private AppComponent mAppComponent;

	public void initAppComponent(Context context) {
		if (mAppComponent == null) {
			mAppComponent = DaggerAppComponent.builder()
			.appModule(new AppModule(context))
			.build();
		}
	}

	public AppComponent getAppComponent() {
		return mAppComponent;
	}
}

File: app/src/main/kotlin/com/github/willjgriff/ethereumwallet/di/AppInjector.kt
package com.github.willjgriff.ethereumwallet.di

import android.content.Context
import com.github.wiljgriff.ethereumwallet.di.DaggerAppComponent
import com.github.willjgriff.ethereumwallet.di.modules.AppModule

/**
 * Created by williamgriffiths on 17/04/2017.
 */
object AppInjector {

    lateinit var appComponent: AppComponent
        set

    fun init(context: Context) {
        appComponent = DaggerAppComponent.builder()
                .appModule(AppModule(context))
                .build()
    }

}